### PR TITLE
Fix QgsCsException messages

### PR DESCRIPTION
**Pull request by nyalldawson**
Heh... for the longest time I thought the useless `forward transform of (inf, inf)` exceptions messages contained within QgsCsException were just proj not giving very useful error messages... but it turns out our QgsCsException message generation logic was completely wonky and generating nonsense messages.

Let's fix that!

Instead of:
```
QgsCsException: forward transform of
(inf, inf)
Error: Fallback transform failed
```

we now get much more useful exceptions, like:

`QgsCsException: Forward transform (EPSG:4326 to EPSG:3857) of (-7603859.000000, -7324441.000000) Error: Invalid coordinate`

### CODE REVIEW

**Comment by nyalldawson** [comment]
@rouault One thing I can't work out how to fix is the hardcoded -61 error number here: https://github.com/qgis/QGIS/blob/master/src/core/proj/qgscoordinatetransform_p.cpp#L335 .

I gather that would have been changed by https://github.com/OSGeo/PROJ/pull/2487, but I'm not 100% sure what the corresponding constant should be. Is it PROJ_ERR_COORD_TRANSFM_NO_OPERATION ?

**Comment by rouault** [comment]
I believe ``if ( errNo && errNo != -61 )`` should just be simplified as `if ( errNo )``  . The PROJ < 8 "-61" code was meant for a generic / unknown error.

**Comment by nyalldawson** [comment]
@rouault ok, removed

File: src/core/proj/qgscoordinatetransform.cpp [hunk 1]
```diff
@@ -829,6 +829,9 @@ void QgsCoordinateTransform::transformCoords( int numPoints, double *x, double *
     }
   }
 
+  // something which won't clash with Proj's real error codes!
+  constexpr int PROJ_RESULT_FALLBACK_OPERATION_FAILED = -481516;
```
Comment (troopa81) [inline src/core/proj/qgscoordinatetransform.cpp:833]: How can we be sure that it won't clash in next releases? Wouldn't be better to add an extra bool ?

Comment (nyalldawson) [reply src/core/proj/qgscoordinatetransform.cpp]: @troopa81 ok, done

**Review by troopa81 — APPROVED** [review]

**Pull request merged**